### PR TITLE
Bookmarks on Mobile

### PR DESCRIPTION
**Pull request by kaitoo1**
### Summary of Changes

This PR adds the Bookmarks Tab to the User Profile on mobile. same behavior as web
- added tab that shows all of your bookmarks
- tab is only visible to you
- added "View Bookmarks" button to bookmark success toast. 
- added Bookmark/Unbookmark option to press+hold context menu on nft preview
- bookmark/unbookmark updates all stored values
- empty state for mobile tab
- scrollable profile tabs if there are 5 tab items

### Demo or Before/After Pics
CleanShot 2024-02-12 at 17 42 59@2x

### Testing Steps 
bookmark and unbookmark items in different places of the app
tap the "View Bookmarks" toast button from different places in the app

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/components/GalleryTabs/GalleryTabBar.tsx [hunk 1]
```diff
@@ -72,23 +72,33 @@ export function GalleryTabBar({
...
-        );
-      })}
+      {/* ScrollView doesnt have a built in way to center the tabs if it doesnt overflow, so only use ScrollView if there are more than 4 tabs */}
+      {routes.length > 4 ? (
+        <ScrollView horizontal showsHorizontalScrollIndicator={false}>
+          {Tabs}
+        </ScrollView>
+      ) : (
+        Tabs
+      )}
```
Comment (jakzaizzat) [inline apps/mobile/src/components/GalleryTabs/GalleryTabBar.tsx:101]: 😍

File: apps/mobile/src/components/ProfileView/ProfileViewBookmarkItem.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  const token = useFragment(
+    graphql`
+      fragment ProfileViewBookmarkItemFragment on Token {
+        dbid
+        definition {
+          community {
+            name
+            ...useNavigateToCommunityScreenFragment
+            # ...getCommunityUrlFromCommunityFragment
```
Comment (jakzaizzat) [inline apps/mobile/src/components/ProfileView/ProfileViewBookmarkItem.tsx:44]: nit: Remove unused fragment

File: apps/mobile/src/components/ProfileView/ProfileViewBookmarkItem.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          onPress={handleCollectionNamePress}
+          eventName="Pressed Collection Name On Bookmarked Token In Bookmarks Tab"
+          eventElementId="Collection Name On Bookmarked Token In Bookmarks Tab"
+          eventContext={contexts.Bookmarks}
+        >
+          <Typography
+            className="text-sm leading-4 text-black dark:text-white h-8 mt-1"
+            numberOfLines={2}
+            font={{ family: 'ABCDiatype', weight: 'Bold' }}
+            style={{}}
```
Comment (jakzaizzat) [inline apps/mobile/src/components/ProfileView/ProfileViewBookmarkItem.tsx:142]: Remove empty styling

File: apps/mobile/src/components/Text.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+export const TitleXS = ({ children }: { children: string }) => {
+  return (
+    <Typography
+      className="text-metal text-xs leading-4 uppercase"
+      font={{ family: 'ABCDiatype', weight: 'Regular' }}
+    >
+      {children}
+    </Typography>
+  );
+};
```
Comment (jakzaizzat) [inline apps/mobile/src/components/Text.tsx:12]: Finally!!

**Review by jakzaizzat — COMMENTED** [review]
The code looks good to me! The bookmarks flow works fine too! 

Have a couple of UI feedbacks

**Comment by jakzaizzat** [comment]
Seems like the empty state didn't show up. Saw the code for empty state was implemented 🤔

CleanShot 2024-02-13 at 19 13 29@2x

File: apps/mobile/src/components/ProfileView/ProfileViewBookmarkItem.tsx (hunk 3 above)
Comment (kaitoo1) [reply apps/mobile/src/components/ProfileView/ProfileViewBookmarkItem.tsx]: good catch, removed

**Comment by kaitoo1** [comment]
thanks for the review @jakzaizzat ! fixed the empty state and odd number width

**Review by jakzaizzat — APPROVED** [review]
LGTM!

**Pull request merged**